File: VB/DateRangeFilterCS/TestData.vb
Imports System
Imports System.Collections.ObjectModel

Namespace DateRangeFilterCS
	Public Class TestData
		Public Property Name() As String
		Public Property Date1() As Date
		Public Property Date2() As Date

		Public Shared ReadOnly Property GetList() As ObservableCollection(Of TestData)
			Get
				Dim list As New ObservableCollection(Of TestData)()
				For i As Integer = 0 To 99
					list.Add(New TestData() With {
						.Name = "Data #" & i,
						.Date1 = Date.Now.AddDays(i),
						.Date2 = (New Date(Date.Today.Year, 1, 1)).AddDays(i)
					})
				Next i
				Return list
			End Get
		End Property
	End Class
End Namespace

File: CS/DateRangeFilterCS/TestData.cs
using System;
using System.Collections.ObjectModel;

namespace DateRangeFilterCS
{
    public class TestData
    {
        public string Name { get; set; }
        public DateTime Date1 { get; set; }
        public DateTime Date2 { get; set; }

        public static ObservableCollection<TestData> GetList {
            get {
                ObservableCollection<TestData> list = new ObservableCollection<TestData>();
                for (int i = 0; i < 100; i++) {
                    list.Add(new TestData() { Name = "Data #" + i, Date1 = DateTime.Now.AddDays(i), Date2 = (new DateTime(DateTime.Today.Year, 1, 1)).AddDays(i) });
                }
                return list;
            }
        }
    }
}